compiler/rustc_trait_selection/src/traits/select/candidate_assembly.rs

```diff
@@ -921,6 +921,12 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         param_env: ty::ParamEnv<'tcx>,
         cause: &ObligationCause<'tcx>,
     ) -> Option<ty::PolyExistentialTraitRef<'tcx>> {
```

Comment:
this function can likely cause a hang. it uses `infcx.predicate_may_hold` inside of the solver :sparkles: